tests/bsim/bluetooth/mesh/src/test_proxy_sol.c

```diff
@@ -0,0 +1,293 @@
+		k_sem_give(&beacon_sem);
+	}
+}
+
+static struct bt_le_scan_cb scan_cb = {
+	.recv = scan_recv,
+};
+
+static void tester_configure(void)
+{
+	uint8_t status;
+
+	k_sem_init(&beacon_sem, 0, 1);
+	bt_le_scan_cb_register(&scan_cb);
+
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &comp);
+
+	ASSERT_OK(bt_mesh_provision(test_net_key, 0, 0, 0, tester_cfg.addr, tester_cfg.dev_key));
+	ASSERT_OK(bt_mesh_cfg_cli_app_key_add(0, tester_cfg.addr, 0, 0, test_app_key, &status));
+}
+
+static void iut_configure(void)
+{
+	uint8_t status;
+
+	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
+	bt_mesh_device_setup(&prov, &comp);
+
+	/* Network stored in flash during immediate_replay_attack */
+	if (!bt_mesh_is_provisioned()) {
+		ASSERT_OK(bt_mesh_provision(test_net_key, 0, 0, 0, iut_cfg.addr, iut_cfg.dev_key));
+	}
+
+	ASSERT_OK(bt_mesh_cfg_cli_app_key_add(0, iut_cfg.addr, 0, 0, test_app_key, &status));
+	ASSERT_OK(bt_mesh_cfg_cli_gatt_proxy_set(0, iut_cfg.addr, BT_MESH_GATT_PROXY_DISABLED,
+						 &status));
+	ASSERT_OK(bt_mesh_priv_beacon_cli_gatt_proxy_set(0, iut_cfg.addr,
+							 BT_MESH_GATT_PROXY_DISABLED, &status));
+	ASSERT_OK(bt_mesh_od_priv_proxy_set(1));
+}
+
+static int sol_fixed_pdu_create(struct bt_mesh_subnet *sub, struct net_buf_simple *pdu)
+{
+	static uint32_t fixed_seq_n = 0x123456;
+
+	net_buf_simple_add_u8(pdu, sub->keys[SUBNET_KEY_TX_IDX(sub)].msg.nid);
+	net_buf_simple_add_u8(pdu, 0x80);
+	net_buf_simple_add_le24(pdu, sys_cpu_to_be24(fixed_seq_n));
+	net_buf_simple_add_le16(pdu, sys_cpu_to_be16(bt_mesh_primary_addr()));
+	net_buf_simple_add_le16(pdu, 0x0000);
+
+	ASSERT_OK(bt_mesh_net_encrypt(&sub->keys[SUBNET_KEY_TX_IDX(sub)].msg.enc, pdu, 0,
+				      BT_MESH_NONCE_SOLICITATION));
+
+	ASSERT_OK(bt_mesh_net_obfuscate(pdu->data, 0,
+					&sub->keys[SUBNET_KEY_TX_IDX(sub)].msg.privacy));
+
+	net_buf_simple_push_u8(pdu, 0);
+	net_buf_simple_push_le16(pdu, BT_UUID_MESH_PROXY_SOLICITATION_VAL);
+
+	return 0;
+}
+
+static int sol_fixed_pdu_send(void)
+{
+	NET_BUF_SIMPLE_DEFINE(pdu, 20);
+	net_buf_simple_init(&pdu, 3);
+
+	struct bt_mesh_subnet *sub = bt_mesh_subnet_find(NULL, NULL);
+
+	uint16_t adv_int = BT_MESH_TRANSMIT_INT(CONFIG_BT_MESH_SOL_ADV_XMIT);
+
+	ASSERT_OK(sol_fixed_pdu_create(sub, &pdu));
+
+	struct bt_data ad[] = {
+		BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+		BT_DATA_BYTES(BT_DATA_UUID16_ALL,
+			      BT_UUID_16_ENCODE(BT_UUID_MESH_PROXY_SOLICITATION_VAL)),
+		BT_DATA(BT_DATA_SVC_DATA16, pdu.data, pdu.size),
+	};
+
+	ASSERT_OK(bt_mesh_adv_bt_data_send(CONFIG_BT_MESH_SOL_ADV_XMIT, adv_int, ad, 3));
+
+	return 0;
+}
+
+static void test_tester_beacon_rcvd(void)
+{
+	tester_configure();
+
+	/* Check that no beacons are currently being picked up by the scanner */
+	ASSERT_EQUAL(-EAGAIN, k_sem_take(&beacon_sem, K_SECONDS(SEM_TIMEOUT)));
+
+	ASSERT_OK(bt_mesh_proxy_solicit(0));
+
+	if (k_sem_take(&beacon_sem, K_SECONDS(SEM_TIMEOUT))) {
```

Comment:
```suggestion
	ASSERT_OK_MSG(k_sem_take(&beacon_sem, K_SECONDS(SEM_TIMEOUT)), "No beacon (proxy advs) reveiced."); 
```

to all `k_sem_take` in tests